docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+
+### CLI default behaviour and overridability
+
+The CLI will have default behavior for each of these mentioned features – however the CLI by default settings should be able to be overridden by providing new/overriden values in the console command (and optionally the advanced configuration .YAML file and response file .rsp placed along with the CLI executable).
+
+# Roadmap: Versions, timeframe and scope
+
+The following is a high-level plan targeting multiple versions, timeframe and scope per each version.
+
+
+![alt text](Images/Plan-Scope.png "Plan and Scope diagram")
+
+The detailed features per scope and version are explained in the following sections.
+
+The conditions for each version are incremental on top of previous versions.
+
+## Gated conditions for v0.1 Private Preview (2019-Q1)
+
+- Supported ML Tasks
+    - Binary Classification
+    - Regression
+
+- NuGet package available at any private NuGet feed ([Azure DevOps?](https://docs.microsoft.com/en-us/azure/devops/artifacts/get-started-nuget?view=azure-devops&tabs=new-nav) [MyGet?](https://docs.myget.org/docs/reference/security)) with daily drops from CI/CD pipelines.
+
+- Add telemetry for the CLI NuGet package. 
+
+- Minimum MVP functionality for users to test custom datasets. Implementation of the CLI arguments especified for v0.1 in the syntax section and code generation section.
+
+- Documentation: Have needed documentation for users to get started autonomously and provide feedback (Installation, getting started and test without help from the dev team).
+
+- Test with ML.NET Samples: Quality of generated code/models targeting the samples should improve the current published samples. 
+
+Nice to have for v0.1:
+
+- Generate code using a data structure class with "vector valued columns" when loading hundreds of columns.
+
+- Implement the `--app-type` argument with 'console' as the by default app.
+
+## Gated conditions for v0.2 Public Preview (2019-H1)
+
+- NuGet package available at the public NuGet feed (major version) and MyGet feed for minor version previews (daily drops).
+- Additional supported ML Tasks
+    - Multi-class classification
+
+- Use of model definition file `--model-settings-file` (.yaml file) for loading model's fine-grain settings. Also, generate the the .yaml file with the infered settings even if a .yaml file was not used, so the user can learn how to customize by using that model definition file and can re-run it, too.
+
+- Use of response file `[<RSP FILE NAME>]` to load the CLI arguments from a file. Also, when user run new, the RSP of current executed commands can be added to the generated project so that it can be re-ran by the user.
+    - Note: Pending to decide if .RSP is needed or the .YAML file is enough.
+
+- Autocomplete for CLI arguments installed automatically.
+
+- Generate the number of "best models" (project folders and file models) specified by `--best-models-count`
+
+- Simple HTML report with minimum models' metrics.
```

Comment:
Random thought, does this have to just be an HTML report? Can it also have an option to output the metrics to the console?

Reply:
Metrics will be on the console. The HTML report would be additional when needed an additional level of analysis, for instance, if comparing multiple models in a chart, etc.
The initial versions of the CLI will only have it on the console, though. 
So, you're right. 👍